src/components/container/Container.tsx

```diff
@@ -0,0 +1,18 @@
+import { ReactNode } from "@tanstack/react-router";
+import React from "react";
+
+interface ContainerProps {
+  children?: ReactNode;
+}
+
+const Container: React.FC<ContainerProps> = React.memo(
```

Comment:
IMO we need to find another name to `Container` as it's too general purpose - do you have any idea?